### PR TITLE
Port connection pool fix from devel back to 3.7.

### PR DESCRIPTION
**Pull request by neunhoef**
This fixes a race in the connection pool which could lead to a connection
being leased and then getting an idle timeout. This leads to sporadic network
request failures under TLS and inefficiencies under TCP.

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11012/

File: 3rdParty/fuerte/src/H1Connection.cpp [hunk 1]
```diff
@@ -405,6 +418,20 @@ void H1Connection<ST>::asyncWriteNextRequest() {
...
+        // 0, if _leased is still 1 (`lease` has happened, but `sendRequest`
+        // has not yet been called), then we leave it untouched. Since
+        // this method here is only ever executed on the iothread and
+        // the value -1 only happens for a brief period in the iothread,
+        // it will never be observed here. Therefore, no actual need to check
+        // the result of the compare_exchange_strong.
+        int exp = 2;
+        if (!_leased.compare_exchange_strong(exp, 0)) {
+          FUERTE_ASSERT(exp != -1);
+        };
```
Comment (jsteemann) [inline 3rdParty/fuerte/src/H1Connection.cpp:434]:
```suggestion
        }
```

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr-windows/4787/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr-windows/4788/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**